formats.go

```diff
@@ -67,7 +67,35 @@ func writeEnvFile(filename string, data map[string]interface{}, mode os.FileMode
 	return writeFile(filename, buf.Bytes(), mode)
 }
 
+func writeCAChain(filename string, data map[string]interface{}, mode os.FileMode) error {
+	const element = "ca_chain"
+	const suffix = "ca"
+
+	// assume this is always a slice so cast the entry to []interface
+	chain, found := data[element].([]interface{})
+	if !found {
+		glog.Errorf("didn't find the certification option: %s", element)
+	}
+
+	name := fmt.Sprintf("%s.%s.%s", filename, element, suffix)
+
+	certChain := ""
+	for count, cert := range chain {
+		certChain += fmt.Sprintf("%s", cert)
+		// append a newline after each cert except last
+		if count < len(chain)-1 {
+			certChain += "\n"
+		}
+	}
+
+	if err := writeFile(name, []byte(fmt.Sprintf("%s", certChain)), mode); err != nil {
+		glog.Errorf("failed to write resource: %s, element: %s, filename: %s, error: %s", filename, suffix, name, err)
+	}
+
+	return nil
+}
 func writeCertificateFile(filename string, data map[string]interface{}, mode os.FileMode) error {
+	writeCAChain(filename, data, mode)
```

Comment:
You're discarding the error response here
```suggestion
	if err := writeCAChain(filename, data, mode); err != nil {
		glog.Errorf("failed to write CA chain: %s", err)
	}
```
Whether you want to continue after failing to write the CA chain is another question.